packages/e2e-tests/helpers/utils.ts

```diff
@@ -173,8 +175,10 @@ export async function waitFor(
 
       return;
     } catch (error) {
-      if (typeof error === "string") {
-        console.log(error);
+      if (!error?.matcherResult && !isString(error)) {
+        // Not an `expect` error and not one of our thrown strings.
+        // → Probably not recoverable.
+        throw error;
```

Comment:
:warning: This is not safe! It means we won't retry on some errors and for that we'd need a *reliable* way to determine if the error is "unrecoverable" (FWIW "recoverable" is not a fitting term here since an error doesn't necessarily mean that something went wrong, it usually only means something that we expect to happen hasn't happened yet).
In particular this would break `waitFor()` if we do FE-2256.

Reply:
→ VC?

Reply:
We now only early-out on `crash` errors.